fedora_messaging/api.py

```diff
@@ -298,6 +298,8 @@ def publish(message, exchange=None, timeout=30):
             message.
         fedora_messaging.exceptions.PublishTimeout: Raised if the broker could not be
             contacted in the given timeout time.
+        fedora_messaging.exceptions.PublishForbidden: Raised if the broker rejects the
+            message because of permission issues.
```

Comment:
I think we should document that users should *always* include the ``except fedora_messaging.exceptions.PublishException`` block in error handling for this call. I can imagine users might not have that and expanding the exception list is an API change (although obviously it breaks in a much worse way without this).

Perhaps something like this in the Raises section:
`fedora_messaging.exceptions.PublishException: The parent class of PublishReturned, PublishTimeout, and PublishForbidden. Any new exceptions raised by this function will sub-class this and it should be handled for compatibility with future releases of fedora-messaging.`

Reply:
I'm not sure the best way to document it, but what I want to convey to the user is their error blocks should be:

```
try:
    api.publish(...)
except PublishTimeout:
    # Special timeout handling
except PublishForbidden:
    # Special forbidden handling
except PublishException:
    # General unexpected exception, either a new exception in the library or the PublishRefused
    # exception we didn't handle explicitly above
```

It's not really related to this change specifically, and I suppose the error is general enough that most applications would prefer to just halt anyway... Maybe it doesn't matter that much.

Reply:
As a library user I think that what I want to know is whether I should retry publishing or just log and drop.